### ISSUE TITLE
DataGrid with groupstyle shows errors

### ISSUE THREAD

**Issue by raytangRT** [MahApps/MahApps.Metro#3236]
## What steps will reproduce this issue?

DataGrid with groupstyle shows the following errors

Related Issue #2356

```
Cannot find source for binding with reference 'RelativeSource FindAncestor, AncestorType='System.Windows.Controls.DataGrid', AncestorLevel='1''. BindingExpression:Path=SelectionUnit; DataItem=null; target element is 'DataGridRow' (Name=''); target property is 'SelectionUnit' (type 'DataGridSelectionUnit')
Cannot find source for binding with reference 'RelativeSource FindAncestor, AncestorType='System.Windows.Controls.DataGrid', AncestorLevel='1''. BindingExpression:Path=AreRowDetailsFrozen; DataItem=null; target element is 'DataGridDetailsPresenter' (Name=''); target property is 'SelectiveScrollingOrientation' (type 'SelectiveScrollingOrientation')
Cannot find source for binding with reference 'RelativeSource FindAncestor, AncestorType='System.Windows.Controls.DataGrid', AncestorLevel='1''. BindingExpression:Path=HeadersVisibility; DataItem=null; target element is 'DataGridRowHeader' (Name=''); target property is 'Visibility' (type 'Visibility')

```
### Expected outcome

No error shown

### Repo

[repo](https://github.com/raytangRT/MahApps.Metro.DataGridGroupStyleError)

### Environment

- MahApps.Metro __v1.6.1__
- Windows OS __Windows 7__
- Visual Studio __2013__
- .NET Framework __4.5__

**Comment by darxis** [comment]
I am also experiencing the following binding error:
`Cannot find source for binding with reference 'RelativeSource FindAncestor, AncestorType='System.Windows.Controls.DataGrid', AncestorLevel='1''. BindingExpression:Path=SelectionUnit; DataItem=null; target element is 'DataGridRow' (Name=''); target property is 'SelectionUnit' (type 'DataGridSelectionUnit')`

when binding the `Visible` property of a `DataGridColumn`

**Comment by lipchev** [comment]
Issue is still here in version **1.6.5**. A workaround that I use is to set the selections manually:

                <DataGrid.RowStyle>
                    <Style TargetType="{x:Type DataGridRow}" BasedOn="{StaticResource AzureDataGridRow}">
                        <Setter Property="mah:DataGridRowHelper.SelectionUnit"
                                Value="{x:Static DataGridSelectionUnit.FullRow}" />
                    </Style>
                </DataGrid.RowStyle>
                <DataGrid.CellStyle>
                    <Style TargetType="{x:Type DataGridCell}" BasedOn="{StaticResource AzureDataGridCell}">
                        <Setter Property="mah:DataGridCellHelper.SelectionUnit"
                                Value="{x:Static DataGridSelectionUnit.FullRow}" />
                    </Style>
                </DataGrid.CellStyle>

**Comment by punker76** [comment]
@raytangRT This is finally fixed now and will be available with v2.0
